python/lsst/ts/observing/block.py

```diff
@@ -0,0 +1,77 @@
+# This file is part of ts_observing.
+#
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+"""
+Classes supporting observing blocks.
+"""
+
+__all__ = ["ObservingBlock", "ObservingScript", "SchedulingConstraints"]
+
+import uuid
+from typing import Any
+
+from pydantic import BaseModel, validator
+
+
+class SchedulingConstraints(BaseModel):
+    """Scheduling constraints to apply to an ObservingBlock."""
+
+    # A key question here is whether different types of constraints are
```

Comment:
Another alternative would be to have`SchedulingConstraints` be a generic representation of a constraint. So a constraint could have `name`, `min_value`, `max_value`.

So one would make:

```
constraints = [
    SchedulingConstraint(
        name="seeing",
        min_value=1.4,
    ),
    SchedulingConstraint(
        name="moon_distance",
        min_value=30.0,
    ),
    SchedulingConstraint(
        name="cloud",
        min_value=0.2,
    ),
]
```

 I think the caveat is that it would not be possible to have a validator I guess...

Reply:
For now I've gone with distinct classes for each type of constraint. This does allow individual validation and I got the serialization to work.